vmm/src/lib.rs

```diff
@@ -1958,124 +2232,10 @@ impl RequestHandler for Vmm {
-                        self.vm_receive_state(&req, &mut socket, mm)?;
-                    } else {
-                        warn!("Configuration not sent yet");
-                        Response::error().write_to(&mut socket)?;
-                    }
-                }
-                Command::Memory => {
-                    info!("Memory Command Received");
-
-                    if !started {
-                        warn!("Migration not started yet");
-                        Response::error().write_to(&mut socket)?;
-                        continue;
-                    }
-                    if let Some(mm) = memory_manager.as_ref() {
-                        self.vm_receive_memory(&req, &mut socket, &mut mm.lock().unwrap())?;
-                    } else {
-                        warn!("Configuration not sent yet");
-                        Response::error().write_to(&mut socket)?;
-                    }
-                }
-                Command::MemoryFd => {
-                    info!("MemoryFd Command Received");
-
-                    if !started {
-                        warn!("Migration not started yet");
-                        Response::error().write_to(&mut socket)?;
-                        continue;
-                    }
-
-                    let mut buf = [0u8; 4];
-                    let (_, file) = socket.recv_with_fd(&mut buf).map_err(|e| {
-                        MigratableError::MigrateReceive(anyhow!(
-                            "Error receiving slot from socket: {}",
-                            e
-                        ))
-                    })?;
-
-                    if existing_memory_files.is_none() {
-                        existing_memory_files = Some(HashMap::default())
-                    }
-
-                    if let Some(ref mut existing_memory_files) = existing_memory_files {
-                        let slot = u32::from_le_bytes(buf);
-                        existing_memory_files.insert(slot, file.unwrap());
-                    }
-
-                    Response::ok().write_to(&mut socket)?;
-                }
-                Command::Complete => {
-                    info!("Complete Command Received");
-                    if let Some(ref mut vm) = self.vm.as_mut() {
-                        vm.resume()?;
-                        Response::ok().write_to(&mut socket)?;
-                    } else {
-                        warn!("VM not created yet");
-                        Response::error().write_to(&mut socket)?;
-                    }
-                    break;
-                }
-                Command::Abandon => {
-                    info!("Abandon Command Received");
-                    self.vm = None;
-                    self.vm_config = None;
-                    Response::ok().write_to(&mut socket).ok();
-                    break;
-                }
-            }
-        }
-
-        Ok(())
+        self.receive_migration(socket)
```

Comment:
Let's not extract this into a new function, otherwise the `vm_receive_migration` function will be a trivial wrapper that contains merely two function calls. Also, keeping the original function will reduce the change size, making the real change clear (such as the changes needed for `Command::MemoryFd`).

Reply:
> Let's not extract this into a new function, otherwise the `vm_receive_migration` function will be a trivial wrapper that contains merely two function calls. Also, keeping the original function will reduce the change size, making the real change clear (such as the changes needed for `Command::MemoryFd`).

I will make the changes as you suggested, thank you for your feedback.